src/pybind/mgr/cephadm/services/iscsi.py

```diff
@@ -142,3 +143,53 @@ def ok_to_stop(self,
         names = [f'{self.TYPE}.{d_id}' for d_id in daemon_ids]
         warn_message = f'It is presumed safe to stop {names}'
         return HandleCommandResult(0, warn_message, '')
+
+    def purge(self, service_name: str) -> None:
+        """Removes configuration
+        """
+        spec = cast(IscsiServiceSpec, self.mgr.spec_store[service_name].spec)
+        try:
+            # remove service configuration from the pool
+            with self.mgr.rados.open_ioctx(spec.pool) as ioctx:
```

Comment:
Next issue: this might hang indefinitely, if the cluster is in trouble. Doesn't sounds like a problem, but users need to be able to fix their cluster using cephadm and this is going to fail, if we're hanging indefinitely here. 

wdyt of testing for HEALTH_OK here? @adk3798

Reply:
I think we should avoid HEALTH_OK tests; better to have a timeout.  librados can do timeouts but it's not super well supported, and this is a shared client instance; it might be better to shell out to the rados CLI to do this.